File: front/src/apis/api/recommendation/RecommendedCourseDto.ts
import {Control} from "../Control";
import {SemesterDto} from "./SemesterDto";
import {SkillDto} from "./SkillDto";

export interface RecommendedCourseDto {
    id: string;
    name: string;
    creditsCount: number;
    semesterEntities: SemesterDto[];
    moduleId: string | null;
    requiredSemesterId: string | null;
    requiredSkills: SkillDto[];
    resultSkills: SkillDto[];
}

File: app/src/main/java/ru/urfu/mm/controller/recommendation/RecommendedCourseDTO.java
package ru.urfu.mm.controller.recommendation;

import java.util.List;
import java.util.UUID;

/**
 * @param requiredSemesterId Если курс является обязательным, то поле заполняется id семестра,
 *                           в котором этот курс будет обязательным, иначе - null.
 */
public record RecommendedCourseDTO(
        UUID id,
        String name,
        int credits,
        List<SemesterDTO> semesters,
        UUID moduleId,
        UUID requiredSemesterId,
        List<SkillDTO> requiredSkills,
        List<SkillDTO> resultSkills
) { }
